include/deal.II/fe/fe_values_coupling.h

```diff
@@ -0,0 +1,557 @@
+     *
+     * @tparam InputVector The @p InputVector type must allow creation of an
+     *   ArrayView object from it; this is satisfied by the `std::vector` class,
+     *   among others.
+     */
+    template <class InputVector>
+    void
+    get_function_values_from_local_dof_values(
+      const InputVector &dof_values,
+      std::vector<solution_value_type<typename InputVector::value_type>>
+        &values) const;
+
+    /**
+     * Return the gradients of the underlying view characterized by
+     * <tt>fe_function</tt> at the renumbered quadrature points.
+     *
+     * This function is the equivalent of the
+     * FEValuesBase::get_function_gradients function but it only works on the
+     * selected view.
+     *
+     * The data type stored by the output vector must be what you get when you
+     * multiply the gradients of shape functions (i.e., @p value_type) times the
+     * type used to store the gradients of the unknowns $U_j$ of your finite
+     * element vector $U$ (represented by the @p fe_function argument).
+     *
+     * @dealiiRequiresUpdateFlags{update_gradients}
+     */
+    template <typename Number>
+    void
+    get_function_gradients(
+      const ReadVector<Number> &                   fe_function,
+      std::vector<solution_gradient_type<Number>> &gradients) const;
+
+    /**
+     * Same as above, but using a vector of renumbered local degree-of-freedom
+     * gradients. In other words, instead of extracting the nodal values of the
+     * degrees of freedom located on the current cell from a global vector
+     * associated with a DoFHandler object (as the function above does), this
+     * function instead takes these local nodal values through its first
+     * argument.
+     *
+     * @param[in] dof_values A vector of local nodal values. This vector must
+     *   have a length equal to the size of the dof renumbering vector.
+     *
+     * @param[out] gradients A vector of gradients of the given finite element
+     *   field, at the renumbered quadrature points on the current object.
+     *
+     * @tparam InputVector The @p InputVector type must allow creation of an
+     *   ArrayView object from it; this is satisfied by the `std::vector` class,
+     *   among others.
+     */
+    template <class InputVector>
+    void
+    get_function_gradients_from_local_dof_values(
+      const InputVector &dof_values,
+      std::vector<solution_gradient_type<typename InputVector::value_type>>
+        &gradients) const;
+
+  private:
+    /**
+     * Helper function that constructs a unique name for a container, based on a
+     * string prefix, on its size, and on the type stored in the container.
+     */
+    template <typename Number>
+    std::string
+    get_unique_container_name(const std::string &prefix,
+                              const unsigned int size,
+                              const Number &     exemplar_number) const;
```

Comment:
The presence of this function surprises me, reading just through the class declaration (not having gotten to the implementation). Can you add to the documentation what the function is used for?

Reply:
ok

Reply:
I've actually moved this in the `RenumberingData` struct, which is now external, and shared by all Views with the same renumbering indices.